File: c/src/Creational/Prototype/soupspoon.c

//SoupSpoon - One of Two Concrete Prototypes extending the AbstractSpoon Prototype

#include "abstract_spoon.h"
#include "soupspoon.h"

spoon_t * soupspoon_new()
{
	spoon_t * s = spoon_new();
	s->name = "Soup Spoon";
}

File: go/creational/prototype/SoupSpoon.go
package prototype


//SoupSpoon - One of Two Concrete Prototypes extending the AbstractSpoon Prototype

type SoupSpoon struct {   // implements Utensil
	Namer
}

func NewSoupSpoon() *SoupSpoon {
	return &SoupSpoon{ Namer{ "Soup Spoon"} }
}
